src/output.rs

```diff
@@ -281,4 +313,68 @@ mod test {
         assert_eq!(&cs.members[2].name, "jwks_uri_x_x");
         assert_eq!(&cs.members[3].name, "jwks_uri_x_x_x");
     }
+
+    #[test]
+    fn can_derive_default() {
+        let containers = vec![
+            Container {
+                name: "Simple".to_string(),
+                level: 1,
+                members: vec![],
+                docs: None,
+                is_enum: false,
+            },
+            Container {
+                name: "Enum".to_string(),
+                level: 1,
+                members: vec![],
+                docs: None,
+                is_enum: true,
+            },
+            Container {
+                name: "Nested".to_string(),
+                level: 1,
+                members: vec![Member {
+                    name: "simple".to_string(),
+                    type_: "Simple".to_string(),
+                    serde_annot: vec![],
+                    extra_annot: vec![],
+                    docs: None,
+                }],
+                docs: None,
+                is_enum: false,
+            },
+            Container {
+                name: "ReferencesEnum".to_string(),
+                level: 1,
+                members: vec![Member {
+                    name: "enum".to_string(),
+                    type_: "Enum".to_string(),
+                    serde_annot: vec![],
+                    extra_annot: vec![],
+                    docs: None,
+                }],
+                docs: None,
+                is_enum: false,
+            },
+            Container {
+                name: "ReferencesEnumNested".to_string(),
+                level: 1,
+                members: vec![Member {
+                    name: "references_enum".to_string(),
+                    type_: "ReferencesEnum".to_string(),
+                    serde_annot: vec![],
+                    extra_annot: vec![],
+                    docs: None,
+                }],
+                docs: None,
+                is_enum: false,
+            },
+        ];
+        assert!(containers[0].can_derive_default(&containers)); // Simple
+        assert!(!containers[1].can_derive_default(&containers)); // Enum
+        assert!(containers[2].can_derive_default(&containers)); // Nested
+        assert!(!containers[3].can_derive_default(&containers)); // ReferencesEnum
+        assert!(!containers[4].can_derive_default(&containers)); // ReferencesEnumNested
```

Comment:
Ah, the search still goes deeply and will exclude anything that includes a raw (non-optioned) enum without a default.

Mind adding one case for a struct containing a `Option<Enum>` because that's kind of the thing we actually are trying to exclude here, but it's not tested.

Reply:
Yes, and `Option<ReferencesEnum>` should also be tested. I'll add that and add a minor perf optimization.